features/filesystem/storagelite/source/StorageLiteFS.h

```diff
@@ -0,0 +1,268 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2018 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef MBED_STORAGELITEFS_H
+#define MBED_STORAGELITEFS_H
+
+#include "FileSystem.h"
+#include "BlockDevice.h"
+#include "StorageLite.h"
+
+/**
+ * StorageLiteFS, FileSystem interface for the StorageLite feature
+ */
+class StorageLiteFS : public mbed::FileSystem {
+public:
+
+    /**
+     * @brief StorageLiteFS constructor.
+     *
+     * @param[in]  name                     Name to add filesystem to tree as.
+     * @param[in]  stlite                   StorageLite instance.
+     * @param[in]  flags                    StorageLite flags given when creating files.
+     *
+     * @return                              none.
+     */
+    StorageLiteFS(const char *name = 0, StorageLite *stlite = 0,
+                  uint32_t flags = StorageLite::encrypt_flag);
```

Comment:
StorageLite flags are proprietary. Having these implemented on a per file basis (in the POSIX open API) would make us add them to the standard flags. This would be misleading as other file systems don't support these flags.

Reply:
Not really a problem if it doesn't conflict with standard Mbed OS flags. The proprietary flags can be provided with a prefix to set them apart.